documentation/adr/0023-error-format.md

```diff
@@ -0,0 +1,74 @@
+_test/shared/matlab_xunit_ISISextras/@TestCaseWithSave/private/instantiate_methods_to_save_.m : TEST_CASE_WITH_SAVE:invalid_argument
+
+erbert_core/classes/MPIFramework/@iMessagesFramework/iMessagesFramework.m                     : iMESSAGES_FRAMEWORK:invalid_argument
+herbert_core/classes/MPIFramework/@iMessagesFramework/iMessagesFramework.m                    : MESSAGES_FRAMEWORK:invalid_argument
+herbert_core/classes/MPIFramework/@iMessagesFramework/private/mix_messages_.m                 : iMESSAGES_FRAMEWOR:invalid_argument
+```
+From which it should be obvious is not a consistent scheme for error identification, even within the same set of routines.
+
+## Decision
+
+As MATLAB standards do not enforce a particular style of error statement,
+this document seeks to outline the format an error identifier should take as well as establish a list of standard common error identifiers.
+
+Following discussions it was decided that it would be useful to have a standard format with 3 components, in the following format
+expanding upon that intially outlined in [ADR 21](./0021-errors-and-warnings.md).
+
+
+```matlab
+error('(HORACE|HERBERT):(function_name|ClassName):error_identifier')
+```
+
+This scheme is based on the current (2020b) MATLAB error identifiers common to many functions.
+
+- `HORACE` or `HERBERT` should be in all caps.
+- The `function_name` and `ClassName` should exactly match that of the parent function/class, including casing.
+- The `error_identifier` should be in `lower_snake` case and should succinctly identify the error message, using one of
+     the following names if the error falls within their remit.
+
+| Identifier         | Issue                                                                                         |
+| :----------------- | :-------------------------------------------------------------------------------------------- |
+| `invalid_argument` | Argument fails validation, there are insufficient arguments or argument is an unexpected flag |
+| `invalid_output`   | Insufficient outputs                                                                          |
+| `not_implemented`  | Called function is virtual/abstract                                                           |
+| `array_mismatch`   | Array dimensions are incompatible (will usually be identified by MATLAB's error)              |
+| `file_not_found`   | File not found on system                                                                      |
+| `io_error`         | Issues with opening, reading or writing files                                                 |
+| `system_error`     | Issues when calling system (shell) functions from MATLAB                                      |
+
+## Consequences
+
+In future error identifiers should be written to follow these guidelines and in cases where surrounding code is being updated an effort should be made to
```

Comment:
There are more consequences:
- errors will be easier to identify in code and at runtime
- legacy errors will need to be updated
- defined standard will help future developers